src/models/users_model.js

```diff
@@ -1,178 +1,204 @@
-    ], callback);
+const storeUser = (client, collection, name, email, hashedPassword, callback) => {
+  collection.insertOne({
+    email,
+    hashedPassword,
+    name,
+  }, (err, result) => {
+    client.close(() => callback(err, result));
+  });
 };
 
-const storeUser = (client, collection, name, email, hashedPassword, callback) => {
-    collection.insertOne({
-        email,
-        hashedPassword,
-        name
-    }, (err, result) => {
-        client.close(() => callback(err, result));
-    });
+const registerUser = (client, collection, name, email, password, callback) => {
+  async.waterfall([
+    next => auth.hashPassword(password, next),
+    (hashedPassword, next) => storeUser(client, collection, name, email, hashedPassword, next),
+  ], callback);
 };
 
 const authorizeUser = (obj, next) => {
-    auth.authorizeLogin(obj.email, obj.password, obj.userInfo, (err, userInfo) => {
-        if (err) {
-            return next(err, null);
-        }
+  auth.authorizeLogin(obj.email, obj.password, obj.userInfo, (err, userInfo) => {
+    if (err) {
+      return next(err, null);
+    }
 
-        const res = {userInfo};
+    const res = { userInfo };
 
-        next(null, res);
-    });
+    return next(null, res);
+  });
 };
 
 const getToken = (obj, next) => {
-    auth.issueToken(obj.userInfo, (err, token) => {
-        next(err, token);
-    });
-};
-
-const fetchUserInfo = ({email, password}, callback, collection) => {
-    collection.findOne({email}, (err, userInfo) => {
-        if (!userInfo) {
-            return callback({message: "username does not exist"});
-        }
-        const res = {
-            email,
-            password,
-            userInfo
-        };
+  auth.issueToken(obj.userInfo, (err, token) => {
+    next(err, token);
+  });
+};
+
+const fetchUserInfo = ({ email, password }, callback, collection) => {
+  collection.findOne({ email }, (err, userInfo) => {
+    if (!userInfo) {
+      return callback({ message: 'username does not exist' });
+    }
+    const res = {
+      email,
+      password,
+      userInfo,
+    };
 
-        callback(err, res);
-    });
+    return callback(err, res);
+  });
 };
 
 const dropUserTable = (client, collection, next) => {
-    collection.drop(() => client.close(next));
+  collection.drop(() => client.close(next));
 };
 
 const createEmailUniqueIndex = (__, collection, next) => {
-    collection.createIndex({"email" : 1}, {unique : true}, next);
+  collection.createIndex({ email: 1 }, { unique: true }, next);
 };
 
 module.exports.clean = (callback) => {
-    async.waterfall([
-        connect,
-        dropUserTable
-    ], callback);
+  async.waterfall([
+    connect, dropUserTable,
+  ], callback);
 };
 
 module.exports.findAllUsers = (callback) => {
-    async.waterfall([
-        connect,
-        selectAllUsers
-    ], callback);
+  async.waterfall([
+    connect, selectAllUsers,
```

Comment:
for the arrays in this file, could we either have it like `[ ...., ...., ]` or
```
[
    ....,
    ....,
]
```

Reply:
Yeah for sure, I'm surprised that lint didn't complain, but I'll make the change :)